fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/AggregateStrategies.java

```diff
@@ -0,0 +1,1234 @@
+
+        List<SlotReference> usedSlotInTable = (List<SlotReference>) (List) Project.findProject(aggUsedSlots,
+                (List<NamedExpression>) (List) olapScan.getOutput());
+
+        for (SlotReference slot : usedSlotInTable) {
+            Column column = slot.getColumn().get();
+            if (keysType == KeysType.AGG_KEYS && !column.isKey()) {
+                return canNotPush;
+            }
+            // The zone map max length of CharFamily is 512, do not
+            // over the length: https://github.com/apache/doris/pull/6293
+            if (mergeOp == PushDownAggOp.MIN_MAX || mergeOp == PushDownAggOp.MIX) {
+                PrimitiveType colType = column.getType().getPrimitiveType();
+                if (colType.isArrayType() || colType.isComplexType() || colType == PrimitiveType.STRING) {
+                    return canNotPush;
+                }
+                if (colType.isCharFamily() && mergeOp != PushDownAggOp.COUNT && column.getType().getLength() > 512) {
+                    return canNotPush;
+                }
+            }
+            if (mergeOp == PushDownAggOp.COUNT || mergeOp == PushDownAggOp.MIX) {
+                // NULL value behavior in `count` function is zero, so
+                // we should not use row_count to speed up query. the col
+                // must be not null
+                if (column.isAllowNull()) {
+                    return canNotPush;
+                }
+            }
+        }
+
+        PhysicalOlapScan physicalOlapScan = (PhysicalOlapScan) new LogicalOlapScanToPhysicalOlapScan()
+                .build()
+                .transform(olapScan, cascadesContext)
+                .get(0);
+
+        return aggregate.withChildren(ImmutableList.of(
+            new PhysicalStorageLayerAggregate(physicalOlapScan, mergeOp)
+        ));
+    }
+
+    /**
+     * sql: select count(*) from tbl group by id
+     *
+     * before:
+     *
+     *          LogicalAggregate(groupBy=[id], output=[count(*)])
+     *                       |
+     *               LogicalOlapScan(table=tbl)
+     *
+     * after:
+     *
+     *  single node aggregate:
+     *
+     *             PhysicalHashAggregate(groupBy=[id], output=[count(*)])
+     *                              |
+     *                 PhysicalDistribute(distributionSpec=GATHER)
+     *                             |
+     *                     LogicalOlapScan(table=tbl)
+     *
+     *  distribute node aggregate:
+     *
+     *            PhysicalHashAggregate(groupBy=[id], output=[count(*)])
+     *                                    |
+     *           LogicalOlapScan(table=tbl, **already distribute by id**)
+     *
+     */
+    private List<PhysicalHashAggregate<Plan>> onePhaseAggregateWithoutDistinct(
+            LogicalAggregate<? extends Plan> logicalAgg, ConnectContext connectContext) {
+        RequireProperties requireGather = RequireProperties.of(PhysicalProperties.GATHER);
```

Comment:
gather require could be a static member of RequireProperties

Reply:
RequireProperties has a `List<PhysicalProperties> properties` match to every require physical properties of children.
so `RequireProperties.GATHER` only use in the one child and not very general